lib/Reader/readerir.cpp

```diff
@@ -3164,65 +3189,72 @@ void GenIR::beginFlowGraphNode(FlowGraphNode *Fg, uint32_t CurrOffset,
   }
 }
 
-// Allow filtering which try regions we support (during EH bring-up)
-static bool hasNYIClause(EHRegion *TryRegion) {
-  for (EHRegionList *ChildNode = rgnGetChildList(TryRegion); ChildNode;
+void GenIR::fgEnterRegion(EHRegion *Region) {
+  if (Region->Kind != ReaderBaseNS::RegionKind::RGN_Try) {
+    // Nothing to do when entering other region kinds -- handler regions get
+    // set up when we enter the try region they handle.
+    return;
+  }
+
+  // Find the outer region handler first.
+  EHRegion *Parent = rgnGetParent(Region);
+  assert(!rgnIsOutsideParent(Region));
+
+  Instruction *EHPad = Parent->HandlerEHPad;
+
+  // When we enter a try region, eagerly create EH pads for its handler(s),
+  // in reverse order, because each needs to name the next as its successor
+  // when it is constructed, and any invokes inside the try need to name the
+  // first handler as their successsor.
+  SmallVector<EHRegion *, 4> Handlers;
+  for (EHRegionList *ChildNode = rgnGetChildList(Region); ChildNode;
```

Comment:
Perhaps we should add an API that allows for reverse iteration of an `EHRegionList`.